modules/adhashBidAdapter.js

```diff
@@ -98,17 +148,50 @@ export const spec = {
   },
 
   buildRequests: (validBidRequests, bidderRequest) => {
+    const storage = getStorageManager({ bidderCode: ADHASH_BIDDER_CODE });
     const { gdprConsent } = bidderRequest;
-    const { url } = spec;
     const bidRequests = [];
     let referrer = '';
-    if (bidderRequest && bidderRequest.refererInfo) {
-      // TODO: is 'page' the right value here?
-      referrer = bidderRequest.refererInfo.page;
+    try {
+      referrer = window.top.location.href;
+    } catch (e) {
+      referrer = window.location.href;
     }
-    for (var i = 0; i < validBidRequests.length; i++) {
-      var index = Math.floor(Math.random() * validBidRequests[i].sizes.length);
-      var size = validBidRequests[i].sizes[index].join('x');
+    const body = document.body;
+    const html = document.documentElement;
+    const pageHeight = Math.max(
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const pageWidth = Math.max(body.scrollWidth, body.offsetWidth, html.clientWidth, html.scrollWidth, html.offsetWidth);
+
+    for (let i = 0; i < validBidRequests.length; i++) {
+      const bidderURL = validBidRequests[i].params.bidderURL || 'https://bidder.adhash.com';
```

Comment:
Can you please ensure that the url passed through `params.bidderURL` is using `https://` and not `http://`.

Reply:
I've added a check in isBidRequestValid as well as unit test for that.